client/src/panels/panel_declarations/finances/detailed_program_spending_split/detailed_program_spending_split.js

```diff
@@ -103,29 +103,44 @@ class HistoricalProgramBars extends React.Component {
       </div>
       <div className="frow">
         <div className="fcol-md-4" style={{ width: "100%" }}>
-          <div
-            className="legend-container"
-            style={{ maxHeight: "400px" }}
-          >
-            <GraphLegend
-              items={_.chain(data)
-                .sortBy(({data}) => _.last(data) || 0 )
-                .reverse()
-                .map( ({ label, id }) => ({
-                  label,
-                  active: _.includes(selected, id),
-                  id,
-                  color: colors(label),
-                }))
-                .value()
-              }
-              onClick={id => {!(selected.length === 1 && selected.includes(id)) &&
-                this.setState({
-                  selected: _.toggle_list(selected, id),
-                });
-              }}
-            />
-          </div>
+          <StandardLegend
+            items={_.chain(data)
+              .sortBy(({data}) => _.last(data) || 0 )
+              .reverse()
+              .map( ({ label, id }) => ({
+                label,
+                active: _.includes(selected, id),
+                id,
+                color: colors(label),
+              }))
+              .value()
+            }
+            onClick={id => {!(selected.length === 1 && selected.includes(id)) &&
+              this.setState({
+                selected: _.toggle_list(selected, id),
+              });
+            }}
+            Controls={
```

Comment:
How about making the redundant component (line 124 to 143) into a reusable component itself somewhere in `legends`, and pass in something like SelectAllOnClick and SelectNoneOnClick ?

Reply:
Sounds good to me!

Reply:
Want to do that before I re-review my self?

Reply:
sure, I'll do that